bridges/snowbridge/primitives/router/src/inbound/tests.rs

```diff
@@ -38,3 +38,37 @@ fn test_contract_location_with_incorrect_location_fails_convert() {
 		None,
 	);
 }
+
+#[test]
+fn test_reanchor_all_assets() {
+	let ethereum_context: InteriorLocation = [GlobalConsensus(Ethereum { chain_id: 1 })].into();
+	let ethereum = Location::new(2, ethereum_context.clone());
+	let ah_context: InteriorLocation = [GlobalConsensus(Polkadot), Parachain(1000)].into();
+	let global_ah = Location::new(1, ah_context.clone());
+	let bh_context: InteriorLocation = [GlobalConsensus(Polkadot), Parachain(1002)].into();
+	let assets = vec![
+		// DOT
+		Location::new(1, []),
+		// GLMR (Some Polkadot parachain currency)
+		Location::new(1, [Parachain(2004)]),
+		// AH asset
+		Location::new(0, [PalletInstance(50), GeneralIndex(42)]),
+		// KSM
+		Location::new(2, [GlobalConsensus(Kusama)]),
+		// KAR (Some Kusama parachain currency)
+		Location::new(2, [GlobalConsensus(Kusama), Parachain(2000)]),
+	];
+	for asset in assets.iter() {
+		// reanchor logic in pallet_xcm on AH
+		let mut reanchored_asset = asset.clone();
+		assert_ok!(reanchored_asset.reanchor(&ethereum, &ah_context));
+		// reanchor back to original location in context of Ethereum
+		let mut reanchored_asset_with_ethereum_context = reanchored_asset.clone();
+		assert_ok!(reanchored_asset_with_ethereum_context.reanchor(&global_ah, &ethereum_context));
+		assert_eq!(reanchored_asset_with_ethereum_context, asset.clone());
+		// reanchor back to original location in context of BH
+		let mut reanchored_asset_with_bh_context = reanchored_asset.clone();
+		assert_ok!(reanchored_asset_with_bh_context.reanchor(&global_ah, &bh_context));
+		assert_eq!(reanchored_asset_with_bh_context, asset.clone());
```

Comment:
I don't think this is done anywhere in the bridge. You have an asset from Ethereum's point of view and reanchor it from the context of bridge hub. `reanchor` shouldn't be used like that. You should always put in `context` the point of view of the asset

Reply:
Thanks! Yeah, the test is just for playing with the reanchor logic and can be removed for less confusion.
https://github.com/paritytech/polkadot-sdk/pull/5546/commits/315070f0fe94975f15c8caed2b3192ab410fd7db